libs/sdk-ui-dashboard/src/dashboard/Dashboard.tsx

```diff
@@ -77,119 +108,63 @@ export interface IDashboardProps {
-                        defaultComponentProps: {
-                            AdditionalMenuItems: [customItem],
-                        },
-                    },
-                },
-            }}
-        />
-    );
-}
+    useEffect(() => {
+        if (!loading && result === undefined) {
+            dispatch(loadDashboard());
+        }
+    }, [loading, result]);
+
+    if (!loading && result === undefined) {
+        return <div>Initializing...</div>;
+    }
 
-const MyCustomAttrFilter: React.FC<IDashboardAttributeFilterProps> = (
-    _props: IDashboardAttributeFilterProps,
-) => {
-    // custom impl.. does whatever it wants, on user selection calls `props.onFilterChanged`
+    if (loading) {
+        return <div>Loading...</div>;
+    }
 
-    return null;
+    if (error) {
+        return <div>Error: {error.message}</div>;
+    }
+
+    return <DashboardInner {...props} />;
 };
 
 /**
- * Shows how to use built-in customization to have custom implementation of filter.
+ * @internal
  */
-export function demonstrateCustomAttributeFilterImpl() {
-    return (
-        <Dashboard
-            filterBarConfig={{
-                defaultComponentProps: {
-                    attributeFilterConfig: {
-                        Component: MyCustomAttrFilter,
-                    },
-                },
-            }}
-        />
-    );
-}
+export const Dashboard: React.FC<IDashboardProps> = (props: IDashboardProps) => {
+    const backend = useBackendStrict(props.backend);
+    const workspace = useWorkspaceStrict(props.workspace);
+    const eventEmitter = createRootEventEmitter(props.eventHandlers);
+
+    const store = createDashboardStore({
+        sagaContext: {
+            backend,
+            workspace,
+            dashboardRef: props.dashboardRef,
+        },
+        rootEventEmitter: eventEmitter.eventEmitterSaga,
+        rootCommandHandler,
+    });
 
-/**
- * Shows how to add a completely custom dashboard rendering.
- */
-export function demonstrateCustomDashboardLayout() {
     return (
-        <Dashboard>
-            {(_dashboard: any) => {
-                return (
-                    <React.Fragment>
-                        <h1>My Custom Dashboard</h1>
-                        <div>
-                            {/* dashboard widget is our component*/}
-                            <DashboardWidget dateDataset={idRef("some.date.dataset")}>
-                                <InsightView insight="something"></InsightView>
-                            </DashboardWidget>
-                        </div>
-                        <div>
-                            <DashboardWidget dateDataset={idRef("some.other.date.dataset")}>
-                                {/* <LineChart ... /> */}
-                            </DashboardWidget>
-                        </div>
-                    </React.Fragment>
-                );
-            }}
-        </Dashboard>
+        <Provider store={store}>
```

Comment:
We should likely create custom context for the store - to avoid clash when the user uses its own redux store ( https://react-redux.js.org/api/provider#props )

Reply:
fixed